tests/ttnn/unit_tests/operations/test_all_gather_matmul.py

```diff
@@ -0,0 +1,233 @@
+    #     tt_output_tensor = t.cpu().to(ttl.tensor.Layout.ROW_MAJOR).to_torch()
+
+    #     eq, output = comp_pcc(tt_output_tensor, matmul_output[i])
+    #     logger.info(f"Output {i}: {output}")
+    #     if not eq:
+    #         logger.error(f"output mismatch for tensor {i}")
+    #     assert eq, f"{i} FAILED: {output}"
+
+    print("Checking outputs for All Gather Matmul (All Gather)")
+    for i, t in enumerate(ttnn.get_device_tensors(tt_all_gather_out_tensor)):
+        tt_output_tensor = t.cpu().to(ttl.tensor.Layout.ROW_MAJOR).to_torch()
+        if input_dtype == ttl.tensor.DataType.BFLOAT16:
+            eq, output = comp_equal(tt_output_tensor, input_tensor)
+        else:
+            eq, output = comp_pcc(tt_output_tensor, input_tensor)
+        logger.info(f"Output {i}: {output}")
+        if not eq:
+            logger.error(f"output mismatch for tensor {i}")
+        assert eq, f"{i} FAILED: {output}"
+
+    print("Checking outputs for All Gather Matmul (Datacopy)")
+    for i, t in enumerate(ttnn.get_device_tensors(tt_datacopy_out_tensor)):
+        tt_output_tensor = t.cpu().to(ttl.tensor.Layout.ROW_MAJOR).to_torch()
+        if input_dtype == ttl.tensor.DataType.BFLOAT16:
+            eq, output = comp_equal(tt_output_tensor, input_tensor)
+        else:
+            eq, output = comp_pcc(tt_output_tensor, input_tensor)
+        logger.info(f"Output {i}: {output}")
+        if not eq:
+            logger.error(f"output mismatch for tensor {i}")
+    assert eq, f"{i} FAILED: {output}"
+
+
+# Enumerate the post-commit cases explicitly
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize(
+    "num_devices, num_links, input_shape, dim, layout, matmul_output_dim",
+    [
+        # (8, 1, [1, 1, 32, 512], 3, ttl.tensor.Layout.TILE, 1024),  # https://github.com/tenstorrent/tt-metal/issues/9686
+        (
+            8,
+            1,
+            [1, 1, 32, 16 * 32],
+            3,
+            ttl.tensor.Layout.TILE,
+            1024,
+        ),
+        (
+            8,
+            1,
+            [1, 1, 128, 128 * 32],
+            3,
+            ttl.tensor.Layout.TILE,
+            1024,
+        ),
+        (
+            8,
+            1,
+            [1, 1, 1024, 1024 * 32],
+            3,
+            ttl.tensor.Layout.TILE,
+            1024,
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "input_dtype",
+    [
+        ttl.tensor.DataType.BFLOAT16,
+        # ttl.tensor.DataType.BFLOAT8_B,          # https://github.com/tenstorrent/tt-metal/issues/9686
+    ],
+)
+@pytest.mark.parametrize(
+    "mem_config",
+    [
+        ttl.tensor.MemoryConfig(buffer_type=ttl.tensor.BufferType.DRAM),
+        # ttl.tensor.MemoryConfig(buffer_type=ttl.tensor.BufferType.L1),  # https://github.com/tenstorrent/tt-metal/issues/9686
+    ],
+)
+@pytest.mark.parametrize("enable_async", [True, False])
+def test_all_gather_on_t3000_post_commit(
```

Comment:
I overlooked this the first time through but this is the same name as the test case in `test_all_gather` - it should be renamed

Reply:
done